src/error.rs

```diff
@@ -25,76 +25,191 @@ macro_rules! acheck {
+        libc::ENODEV => "No such device",
+        libc::ENOTDIR => "Not a directory",
+        libc::EISDIR => "Is a directory",
+        libc::EINVAL => "Invalid argument",
+        libc::ENFILE => "File table overflow",
+        libc::EMFILE => "Too many open files",
+        libc::ENOTTY => "Not a typewriter",
+        libc::ETXTBSY => "Text file busy",
+        libc::EFBIG => "File too large",
+        libc::ENOSPC => "No space left on device",
+        libc::ESPIPE => "Illegal seek",
+        libc::EROFS => "Read-only file system",
+        libc::EMLINK => "Too many links",
+        libc::EPIPE => "Broken pipe",
+        libc::EDOM => "Math argument out of domain of func",
+        libc::ERANGE => "Math result not representable",
+        libc::EDEADLK => "Resource deadlock would occur",
+        libc::ENAMETOOLONG => "File name too long",
+        libc::ENOLCK => "No record locks available",
+        libc::ENOSYS => "Function not implemented",
+        libc::ENOTEMPTY => "Directory not empty",
+        libc::ELOOP => "Too many symbolic links encountered",
+        libc::ENOMSG => "No message of desired type",
+        libc::EIDRM => "Identifier removed",
+        libc::EINPROGRESS => "Operation now in progress",
+        libc::EALREADY => "Operation already in progress",
+        libc::ENOTSOCK => "Socket operation on non-socket",
+        libc::EDESTADDRREQ => "Destination address required",
+        libc::EMSGSIZE => "Message too long",
+        libc::EPROTOTYPE => "Protocol wrong type for socket",
+        libc::ENOPROTOOPT => "Protocol not available",
+        libc::EPROTONOSUPPORT => "Protocol not supported",
+        libc::ESOCKTNOSUPPORT => "Socket type not supported",
+        libc::EPFNOSUPPORT => "Protocol family not supported",
+        libc::EAFNOSUPPORT => "Address family not supported by protocol",
+        libc::EADDRINUSE => "Address already in use",
+        libc::EADDRNOTAVAIL => "Cannot assign requested address",
+        libc::ENETDOWN => "Network is down",
+        libc::ENETUNREACH => "Network is unreachable",
+        libc::ENETRESET => "Network dropped connection because of reset",
+        libc::ECONNABORTED => "Software caused connection abort",
+        libc::ECONNRESET => "Connection reset by peer",
+        libc::ENOBUFS => "No buffer space available",
+        libc::EISCONN => "Transport endpoint is already connected",
+        libc::ENOTCONN => "Transport endpoint is not connected",
+        libc::ESHUTDOWN => "Cannot send after transport endpoint shutdown",
+        libc::ETOOMANYREFS => "Too many references: cannot splice",
+        libc::ETIMEDOUT => "Connection timed out",
+        libc::ECONNREFUSED => "Connection refused",
+        libc::EHOSTDOWN => "Host is down",
+        libc::EHOSTUNREACH => "No route to host",
+        libc::ENOTSUP => "Operation not supported",
+        _ => "Unknown errno",
+    }
 }
 
+impl From<Error> for fmt::Error {
+    fn from(_: Error) -> fmt::Error {
+        fmt::Error
```

Comment:
https://github.com/diwic/alsa-rs/blob/db4bdd7f4e467148c091720589d217fdafb5d168/src/chmap.rs#L57 and https://github.com/diwic/alsa-rs/blob/db4bdd7f4e467148c091720589d217fdafb5d168/src/chmap.rs#L84 convert to a `fmt::Error`, normally I would have refactored this to avoid losing the error information but I wanted to keep the changes minimal.